pkg/services/ngalert/schedule/alert_rule_test.go

```diff
@@ -762,8 +763,95 @@ func TestRuleRoutine(t *testing.T) {
 
 		require.NotEmpty(t, sch.stateManager.GetStatesForRuleUID(rule.OrgID, rule.UID))
 	})
+
+	t.Run("when there are resolved alerts they should keep sending until retention period is over", func(t *testing.T) {
```

Comment:
NOTE: Main test showing multiple sends for a resolved notification and how it interacts with resend delay.